packages/analytics-node-test/test/index.test.ts

```diff
@@ -0,0 +1,439 @@
+import * as amplitude from '@amplitude/analytics-node';
+import { default as nock } from 'nock';
+import { success } from './responses';
+import { path, SUCCESS_MESSAGE, url, uuidPattern } from './constants';
+
+describe('integration', () => {
+  const uuid: string = expect.stringMatching(uuidPattern) as string;
+  const library = expect.stringMatching(/^amplitude-node-ts\/.+/) as string;
+  const number = expect.any(Number) as number;
+
+  describe('track', () => {
+    test('should track event', async () => {
+      const scope = nock(url).post(path).reply(200, success);
+
+      await amplitude.init('API_KEY');
```

Comment:
should this have a `.promise` if we are awaiting on it?

Reply:
oh i just realized the promise interface for node client is different from browser client. is this intentional? what's he reason for having a different interface?

cc @justin-fiedler

Reply:
If I remember correctly, browser has `.promise` because it has snippet and snippet needs a promise. NodeJS does not need `.promise` because it returns a promise type.

Reply:
So`.promise` interface is actually unrelated to having a snippet and I believe snippet would have worked regardless of the interface. The decision to use `.promise` is to avoid warnings saying ignored promise in the most common usage:

```ts
track('event')
```

To get around this warning one either has to do

```ts
await track('event')
```

or 

```ts
void track('event')
```

Most analytics usages won't use `await` since it shouldn't be blocking, while adding `void` seemed unnecessary.

To resolve this, our interface was updated so that `track()` or other awaitable fns do not return a promise, but rather an object with the following interface:

```ts
{
  promise: Promise<T>
}
```

This allows most common usage to not display warnings on ignored promise because track('') does not directly return a Promise.

```ts
track('event')
```

If user want to await, then track would look like:

```
await track('event').promise
```

Reply:
Interesting case. I will add `.promise` back to methods.